src/rules/no-missing-label-refs.js

```diff
@@ -33,17 +33,18 @@ import { findOffsets, illegalShorthandTailPattern } from "../util.js";
  * @returns {Array<{label:string,position:Position}>} The missing references.
  */
 function findMissingReferences(node, nodeText) {
+	/** @type {Array<{label:string,position:Position}>} */
 	const missing = [];
 	const nodeStartLine = node.position.start.line;
 	const nodeStartColumn = node.position.start.column;
 
-	/*
-	 * Matches substrings like "[foo]", "[]", "[foo][bar]", "[foo][]", "[][bar]", or "[][]".
+	/**
+	 * Matches substrings like `"[foo]"`, `"[]"`, `"[foo][bar]"`, `"[foo][]"`, `"[][bar]"`, or `"[][]"`.
 	 * `left` is the content between the first brackets. It can be empty.
 	 * `right` is the content between the second brackets. It can be empty, and it can be undefined.
 	 */
 	const labelPattern =
-		/(?<!\\)\[(?<left>(?:\\.|[^[\]])*)(?<!\\)\](?:\[(?<right>(?:\\.|[^\]])*)(?<!\\)\])?/dgu;
+		/(?<=(?<!\\)(?:\\{2})*)\[(?<left>(?:\\.|[^[\]])*(?<!\\)(?:\\{2})*)\](?:\[(?<right>(?:\\.|[^\]])*(?<!\\)(?:\\{2})*)\])?/dgu;
```

Comment:
The pattern around `]` is different from `(?<=(?<!\\)(?:\\{2})*)` at the beginning of the regex, as it was reported by `eslint-plugin-regexp` for the following reason.

<img width="761" height="91" alt="image" src="https://github.com/user-attachments/assets/6a425938-05d2-4bca-b9c6-e620423290bb" />

Reply:
Maybe we could simplify it this way?

```suggestion
		/(?<=(?<!\\)(?:\\{2})*)\[(?<left>(?:\\.|[^[\]\\])*)\](?:\[(?<right>(?:\\.|[^\]\\])*)\])?/dgu;
```

Reply:
Looks so good! I've added a new commit https://github.com/eslint/markdown/pull/490/commits/cb70137ca6c6676e017d403284bfaebefe01f7d7